### PR TITLE
Fix CommanderAgent

### PR DESCRIPTION
**Pull request by brianpetro**
Not working when it is scheduled or manually run(`check` function typo).

### CODE REVIEW

**Comment by virtadpt** [comment]
...so it was a bug!  I thought I'd screwed up some of my detection and triggering logic elsewhere in the agent network!

**Comment by brianpetro** [comment]
Just fixed a similar bug in the SchedulerAgent.

Also, made it so that CommanderAgent can "configure" options containing nested hashes(see DataOutputAgent).

**Comment by cantino** [comment]
Both good catches, thank you for fixing these!

File: app/models/agents/scheduler_agent.rb [hunk 1]
```diff
@@ -87,7 +87,7 @@ def working?
       true
     end
 
-    def check!
```
Comment (knu) [inline app/models/agents/scheduler_agent.rb]: No, this was correct.  SchedulerAgent is only invoked by lib/huginn_scheduler.rb via the `check!` method and the  `check` method is intentionally left undefined.

File: app/models/agents/commander_agent.rb [hunk 2]
```diff
@@ -36,7 +36,7 @@ def working?
       true
     end
 
-    def check!
+    def check
```
Comment (knu) [inline app/models/agents/commander_agent.rb]: You are right.  Thanks for the catch.

File: app/models/agents/scheduler_agent.rb (hunk 1 above)
Comment (knu) [inline app/models/agents/scheduler_agent.rb]: I can't recall what was the reason for the name, but I guess we can now remove `check!` here and make huginn_scheduler directly call `control!` instead.

Comment (cantino) [inline app/models/agents/scheduler_agent.rb]: Oh! Do you want to revert this?

**Pull request merged**